### PR TITLE
Boilerplate/164 seed more usefull data

### LINKED ISSUE #164: Seed more usefull data in Seeder
# Description
Configure seeder to seed more data usefull for development.

# DoD
- It creates more data such as: Users of different roles, more titles and books of said titles, loans.
- It should add fixtures to test folder.

# Context
Blocked by #163
Relates to #145

### PR DESCRIPTION
**Pull request by TE4-emanuel-strom**
## What issue did you implement or fix?
Fixed #164

## Summary
Now seeds:
- 4 titles, two of each type
- A bunch of books from different titles which have either status ok or broken
- A loan
Also dumps fixtures.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## How can this implementation be improved?
- Right now the seeded data should be working with current tests in place, they do not but I am quite sure that is the tests fault... Data might have to be tweeked in order for future/ improved tests to pass.

### CODE REVIEW

File: backend/app/controllers/api/v1/loans_controller.rb [hunk 1]
```diff
@@ -9,13 +9,14 @@ def show
...
   end
-
+  
   def create
     @loan = Loan.new(loan_params)
     authorize @loan
     if @loan.save
       render json: @loan
     else
+      pp loan_params
```
Comment (TE4-jacob-lundberg) [inline backend/app/controllers/api/v1/loans_controller.rb]: Is this print necessary?

File: backend/test/fixtures/books.yml [hunk 2]
```diff
@@ -2,37 +2,9 @@
 book_5000:
   title_id: 1
   barcode: '5000'
-  status: Broken
```
Comment (TE4-jacob-lundberg) [inline backend/test/fixtures/books.yml]: This seems to just replace the old books in the fixture with a single one? Is this file generated with more books when $ bin/rails db:seed is run?

File: backend/test/controllers/users_controller_test.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 class UsersControllerTest < ActionDispatch::IntegrationTest
   setup do
-    @user = users(:user_1904583927)
+    @user = users(:user_1954282603)
```
Comment (TE4-jacob-lundberg) [inline backend/test/controllers/users_controller_test.rb]: Are you supposed to change the user in the test? Might this be what causes the errors to fail?

File: backend/Gemfile [hunk 4]
```diff
@@ -40,6 +40,8 @@ gem 'bootsnap', '>= 1.1.0', require: false
 
 gem "pundit"
 gem "rack-cors"
+gem 'faker'
```
Comment (TE4-jacob-lundberg) [inline backend/Gemfile]: Did you need to add faker to two gemfiles?

File: backend/db/seeds.rb [hunk 5]
```diff
@@ -6,22 +6,40 @@
...
+title_alice = Title.create(name: "Alice in Wonderland", isbn: "97161949222", cost: 70, title_type: "Skönlitteratur")
+title_lotr = Title.create(name: "The Fellowship of the Ring", isbn: "9780547928210", cost: 100, title_type: "Skönlitteratur")
 
-book = Book.create(barcode: "5000", status: "Broken", title: title)
+title_cc = Title.create(name: "Clean Code", isbn: "9780132350884", cost: 300, title_type: "Kurslitteratur")
+title_tomtens_jul = Title.create(name: "Tomtens jul", isbn: "9781999985462", cost: 60, title_type: "Kurslitteratur")
+
+# Create books to be loaned
+book_alice = Book.create(barcode: 5000, status: "OK", title: title_alice)
+book_cc = Book.create(barcode: 1001, status: "OK", title: title_cc)
```
Comment (itggot-linus-styren) [inline backend/db/seeds.rb]: ~~There isn't any loan for this book AFAICT~~, maybe there should be a returned loan?

Comment (itggot-linus-styren) [reply backend/db/seeds.rb]: Perhaps you can change the book used in the unit test to one of those declared below.

File: backend/test/controllers/users_controller_test.rb (hunk 3 above)
Comment (TE4-emanuel-strom) [reply backend/test/controllers/users_controller_test.rb]: This is an intentional change, the new user is 'låneservice johanneberg'

File: backend/db/seeds.rb [hunk 6]
```diff
@@ -6,22 +6,60 @@
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
 
-title = Title.create(name: "Programming for dummies", isbn: "420-420", cost: 200, title_type: "Skönlitteratur")
+# Import Users
+ActiveSupport.on_load(:active_job) do
```
Comment (TE4-jacob-lundberg) [inline backend/db/seeds.rb]: Nice, you managed to fetch all users within the seeder now

**Comment by te4-david-petersson** [comment]
@TE4-jacob-lundberg can this be merged now?

**Pull request merged**